peekingduck/pipeline/nodes/draw/tag.py

```diff
@@ -13,48 +13,158 @@
+    thus each attribute consists of a list with 2 tags. ::
+
+        # Example
+        {"obj_attrs": {"ids":[1,2], "details": {"gender": ["female","male"], "age": [52,17]}}
+
+    The table below illustrates how ``show`` can be configured to achieve different outcomes for
+    this example. Key takeaways are:
+
+    * To draw nested attributes, include all the keys leading to them (within the ``obj_attrs`` \
+    dictionary), separating each key with a ``->``. \
+
+    * To draw multiple comma-separated attributes above each bounding box, add them to the list \
+    of ``show`` config.
+
+    +-----+-----------------------------------------+---------------+---------------+
+    | No. | ``show`` config                         | Tag above 1st | Tag above 2nd |
+    |     |                                         | bounding box  | bounding box  |
+    +-----+-----------------------------------------+---------------+---------------+
+    | 1.  | ["ids"]                                 | "1"           | "2"           |
+    +-----+-----------------------------------------+---------------+---------------+
+    | 2.  | ["details -> gender"]                   | "female"      | "male"        |
+    +-----+-----------------------------------------+---------------+---------------+
+    | 3.  | ["details -> age", "details -> gender"] | "52, female"  | "17, male"    |
+    +-----+-----------------------------------------+---------------+---------------+
 
     Inputs:
         |img|
 
         |bboxes|
 
-        |obj_tags|
+        |obj_attrs|
 
     Outputs:
         |no_output|
 
     Configs:
-        None.
+        show (:obj:`List[str]`): **default = []**. |br|
+            List the desired attributes to be drawn. For more details on how to use this config,
+            see the section above.
+        tag_color (:obj:`List[int]`): **default = [77, 103, 255]**. |br|
+            Define the color of the drawn tag, in BGR format. Defined values have to be integers,
+            and 0 <= value <= 255.
+
+    .. versionchanged:: 1.2.0 |br|
+        :mod:`draw.tag` used to take in ``obj_tags`` (:obj:`List[str]`) as an input data type,
+        which has been deprecated and now subsumed under ``obj_attrs`` (:obj:`Dict[str, Any]`),
+        giving this node more flexibility. Also, the ``tag_color`` config is added to provide
+        the option of changing the tag's color.
     """
 
     def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
+        self.attr_keys = []
+        check_bgr_type(self.tag_color)
+        if not self.show:
+            raise KeyError(
+                "The 'show' config is currently empty. Add the desired attributes to be drawn "
+                "to the list, in order to proceed."
+            )
+        for attr in self.show:
+            attr = attr.replace(" ", "")
+            self.attr_keys.append(re.split(r"->", attr))
```

Comment:
Missed this earlier: using `str.split()` is very much faster than `re.split()` here since `->` is fixed.
In general, `re.` calls should only be used if there is a need for regular expressions, due to the inherent overhead.

Reply:
Updated accordingly.